src/main/java/de/tum/in/www1/artemis/service/AuthorizationCheckService.java

```diff
@@ -483,10 +484,41 @@ public boolean isAllowedToSeeLectureUnit(@NotNull LectureUnit lectureUnit, @Null
         return isAtLeastTeachingAssistantInCourse(course, user) || (isStudentInCourse(course, user) && lectureUnit.isVisibleToStudents());
     }
 
+    /**
+     * Determines if a user is allowed to see private information about a tutorial group such as the list of registered students
+     *
+     * @param tutorialGroup the tutorial group for which to check permission
+     * @param user          the user for which to check permission
+     * @return true if the user is allowed, false otherwise
+     */
+    public boolean isAllowedToSeePrivateTutorialGroupInformation(@NotNull TutorialGroup tutorialGroup, @Nullable User user) {
+        if (user == null || user.getGroups() == null) {
+            user = userRepository.getUserWithGroupsAndAuthorities();
+        }
+        if (isAdmin(user)) {
+            return true;
+        }
+        Course course = tutorialGroup.getCourse();
+        return isAtLeastEditorInCourse(course, user) || (tutorialGroup.getTeachingAssistant() != null && tutorialGroup.getTeachingAssistant().equals(user));
```

Comment:
The role of editors is to help instructors to create exercises. Should they also be allowed to see these private information?

Reply:
I discussed this with @krusche and we decided that only instructors and the tutor should be able to see this and register / deregister students. So I implemented this in 677a9806c5f068adfdf9d2689df4006b3eb09206